clang/lib/CodeGen/CodeGenTBAA.cpp

```diff
@@ -196,11 +196,14 @@ llvm::MDNode *CodeGenTBAA::getTypeInfoHelper(const Type *Ty) {
   // Enum types are distinct types. In C++ they have "underlying types",
   // however they aren't related for TBAA.
   if (const EnumType *ETy = dyn_cast<EnumType>(Ty)) {
+    if (!Features.CPlusPlus)
+      return getTypeInfo(Context.getIntTypeForBitwidth(Size * 8, 0));
```

Comment:
I am not sure if this is entirely correct so would appreciate some guidance here!

Reply:
Maybe `ETy->getDecl()->getIntegerType()` ?

Reply:
Good suggestion - thanks!